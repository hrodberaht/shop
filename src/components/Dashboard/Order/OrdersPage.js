import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { getOrders, getLoadedStatus } from '../../../store/orders/selectors';
import ConnectedAuthorization from '../../Auth/Authorization';

import { fetchOrders, fetchChangeOrderStatus } from '../../../store/orders/actionCreators';
import { getAuthToken, getAuthUserId } from '../../../store/authenticate/selectors';
import OrdersSearch from './OrdersSearch';
import OrdersList from './OrdersList';

export class OrdersPage extends Component {
  state = {
    status: 'all',
  };

  componentDidMount() {
    const { getOrdersFromServer, token, userId } = this.props;
    getOrdersFromServer(userId, token);
  }

  handleSelect = (e) => {
    this.setState({ status: e.target.value });
  };

  render() {
    const { orders, loaded } = this.props;
    if (!loaded) return <p>Loading...</p>;
    return (
      <React.Fragment>
        <div>
          <OrdersSearch handleSelect={this.handleSelect} />
        </div>
        <table>
          <tbody>
            <tr className="order-table-title">
              <th>Id:</th>
              <th>Person:</th>
              <th>Date:</th>
              <th>Total price:</th>
              <th>Status:</th>
              <th>Actions:</th>
              <ConnectedAuthorization render withRoleAdmin={<th>Admin:</th>} />
            </tr>
            <OrdersList filters={this.state.status} orders={orders} />
          </tbody>
        </table>
      </React.Fragment>
    );
  }
}
const mapStateToProps = state => ({
  token: getAuthToken(state),
  orders: getOrders(state),
  loaded: getLoadedStatus(state),
  userId: getAuthUserId(state),
});

export default connect(
  mapStateToProps,
  {
    handleClick: fetchChangeOrderStatus,
    getOrdersFromServer: fetchOrders,
  },
)(OrdersPage);

OrdersPage.propTypes = {
  orders: PropTypes.arrayOf(PropTypes.object).isRequired,
  loaded: PropTypes.bool.isRequired,
  getOrdersFromServer: PropTypes.func.isRequired,
  token: PropTypes.string,
  userId: PropTypes.string,
};

OrdersPage.defaultProps = {
  token: null,
  userId: null,
};
