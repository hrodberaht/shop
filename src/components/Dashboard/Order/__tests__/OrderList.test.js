import { OrdersPage } from '../OrdersPage';

const defaultProps = {
  orders: [
    {
      userId: 'user-3',
      person: 'Ben Morrison',
      totalPrice: 2600,
      status: 'realized',
      date: '2018-11-02T18:03:36.535Z',
      companyId: 'company-2',
      orderPositionIds: ['Ld6GsMELf', 'IQW79vU4W', '4VgGGnrnV'],
      id: 'e1oPVJh',
      productsOrder: [
        {
          productId: 'product-1',
          name: 'ZX3',
          price: 400,
          pcsOrder: 1,
          totalPrice: 400,
          id: 'Ld6GsMELf',
        },
        {
          productId: 'product-2',
          name: 'ZX5',
          price: 600,
          pcsOrder: 1,
          totalPrice: 600,
          id: 'IQW79vU4W',
        },
        {
          productId: 'product-3',
          name: 'ZX1',
          price: 200,
          pcsOrder: 8,
          totalPrice: 1600,
          id: '4VgGGnrnV',
        },
      ],
    },
  ],
  getOrdersFromServer: jest.fn(),
  handleClick: jest.fn(),
  loaded: false,
  token: '',
};
const setup = buildSetup(OrdersPage, defaultProps);

describe('<OrderList />', () => {
  it('should render without crash', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
  it('should render if loaded true ', () => {
    const { wrapper } = setup();
    wrapper.setProps({ loaded: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('toggleUnrealizedOrders should change state after call', () => {
    const { wrapper } = setup();
    wrapper.instance().toggleUnrelizedOrders();
    expect(wrapper.state()).toEqual({ isOnlyUnrealizedOrders: true });
    wrapper.instance().toggleUnrelizedOrders();
    expect(wrapper.state()).toEqual({ isOnlyUnrealizedOrders: false });
  });
});
