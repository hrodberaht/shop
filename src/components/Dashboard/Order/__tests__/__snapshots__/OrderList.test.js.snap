// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderList /> should render if loaded true  1`] = `
<Fragment>
  <div>
    <OrdersSearch
      handleChange={[Function]}
      handleSelect={[Function]}
    />
    <hr />
  </div>
  <table>
    <tbody>
      <tr
        className="order-table-title"
      >
        <th>
          Id:
        </th>
        <th>
          Person:
        </th>
        <th>
          Date:
        </th>
        <th>
          Total price:
        </th>
        <th>
          Status:
        </th>
        <th>
          Actions:
        </th>
        <Connect(Authorization)
          render={true}
          withRoleAdmin={
            <th>
              Admin:
            </th>
          }
        />
      </tr>
      <Connect(OrdersList)
        filterValues={
          Object {
            "id": "",
            "status": "all",
          }
        }
        orders={
          Array [
            Object {
              "companyId": "company-2",
              "date": "2018-11-02T18:03:36.535Z",
              "id": "e1oPVJh",
              "orderPositionIds": Array [
                "Ld6GsMELf",
                "IQW79vU4W",
                "4VgGGnrnV",
              ],
              "person": "Ben Morrison",
              "productsOrder": Array [
                Object {
                  "id": "Ld6GsMELf",
                  "name": "ZX3",
                  "pcsOrder": 1,
                  "price": 400,
                  "productId": "product-1",
                  "totalPrice": 400,
                },
                Object {
                  "id": "IQW79vU4W",
                  "name": "ZX5",
                  "pcsOrder": 1,
                  "price": 600,
                  "productId": "product-2",
                  "totalPrice": 600,
                },
                Object {
                  "id": "4VgGGnrnV",
                  "name": "ZX1",
                  "pcsOrder": 8,
                  "price": 200,
                  "productId": "product-3",
                  "totalPrice": 1600,
                },
              ],
              "status": "realized",
              "totalPrice": 2600,
              "userId": "user-3",
            },
          ]
        }
      />
    </tbody>
  </table>
</Fragment>
`;

exports[`<OrderList /> should render without crash 1`] = `
<p>
  Loading...
</p>
`;
